src/database/api/Database.ts

```diff
@@ -0,0 +1,144 @@
+import { fatal } from "../core/util/util";
+import { parseRepoInfo } from "../core/util/libs/parser";
+import { Path } from "../core/util/Path";
+import { PromiseImpl } from "../../utils/promise";
+import { Reference } from "./Reference";
+import { Repo } from "../core/Repo";
+import { RepoManager } from "../core/RepoManager";
+import { validateArgCount } from "../../utils/validation";
+import { validateUrl } from "../core/util/validation";
+
+/**
+ * Class representing a firebase database.
+ * @implements {firebase.Service}
+ */
+export class Database {
+  /** @type {Repo} */
+  repo_;
```

Comment:
FYI there's another PR (#66) that builds on top of this one adding a whole bunch of types, const/let, private, etc.